backend/app/core/bot.py

```diff
@@ -95,3 +95,16 @@ async def create_room_to_external_url_mapping(self):
                 self.room_to_external_url_mapping[value.room_id].temporary.add(url_code)
             else:
                 self.room_to_external_url_mapping[value.room_id].permanent = url_code
+
+    async def get_rooms_with_mod_permissions(self, user_id: str):
+        rooms_with_mod_permissions = {}
+        for room_id, room_object in self.rooms.items():
+            # Check if bot has permissions to kick and invite
+            if room_object.power_levels.can_user_invite(
+                self.user_id
+            ) and room_object.power_levels.can_user_kick(self.user_id):
+                # Check if user is a member of the room
+                if user_id in room_object.users and not room_object.users[user_id].invited:
+                    if room_object.power_levels.get_user_level(user_id) >= 50:
```

Comment:
Slight concern from my side that we're expecting moderation permissions to always be 50 (which is often, but not always the case). I wonder if we should instead look to something like `state_default` as a guideline for what's considered powerful. This might be overkill though.

@Cadair ?

Reply:
I've added a new option in the `config.sample.yml` to edit the minimum power level required i.e. `min_power_level`.
It's in PR #39 under commit [9c83156](https://github.com/MatrixDoctors/matrix-cerberus/pull/39/commits/9c831560771d3b62aee293fdfb1631a5b9c6265d).

https://github.com/MatrixDoctors/matrix-cerberus/blob/9c831560771d3b62aee293fdfb1631a5b9c6265d/backend/config.sample.yml#L17-L21

Reply:
Added an issue to track the generic feature, don't think it's worth holding things up here: https://github.com/MatrixDoctors/matrix-cerberus/issues/45